core/router.js

```diff
@@ -0,0 +1,62 @@
+define('core/router', [
```

Comment:
this file could use some comments (docstring + what's happening in addRoute). requires don't need to be on different lines.